presentation/src/main/java/com/anytypeio/anytype/presentation/sets/ObjectSetExtension.kt

```diff
@@ -370,6 +375,31 @@ fun ObjectState.DataView.filterOutDeletedAndMissingObjects(query: List<Id>): Lis
     return query.filter(::isValidObject)
 }
 
+suspend fun ObjectState.DataView.Set.isTemplatesAllowed(
+    setOfValue: List<Id>,
+    storeOfObjectTypes: StoreOfObjectTypes,
+    getDefaultPageType: GetDefaultPageType
+): Boolean {
+    val objectDetails = details[setOfValue.first()]?.map.orEmpty()
+    return when (objectDetails.type){
+        ObjectTypeIds.OBJECT_TYPE -> {
+            val objectWrapper = ObjectWrapper.Type(objectDetails)
+            objectWrapper.isTemplatesAllowed()
+        }
+        ObjectTypeIds.RELATION -> {
+            //We have set of relations, need to check default object type
+            storeOfObjectTypes.isTemplatesAllowedForDefaultType(getDefaultPageType)
+        }
+        else -> false
+    }
+}
+
+suspend fun StoreOfObjectTypes.isTemplatesAllowedForDefaultType(getDefaultPageType: GetDefaultPageType): Boolean {
```

Comment:
please add throws annotation here, since you're using here a use-case which might throw an exception. consider handling an exception here to make this method safe

Reply:
done